src/spaceObjects/cpuShip.cpp

```diff
@@ -213,6 +214,20 @@ void CpuShip::orderDock(P<SpaceObject> object)
     this->addBroadcast(FVF_Friendly, "Docking to " + object->getCallSign() + ".");
 }
 
+void CpuShip::orderRepair()
+{
+    if (orders != AI_Repair)
+    {
+        previous_orders = orders;
+        orders = AI_Repair;
+    }
+}
+
+void CpuShip::orderResume()
+{
+    orders = previous_orders;
```

Comment:
There might need to be a fallback here in case the previous orders are no longer relevant (such as defending a ship or object that no longer exists).